### ISSUE TITLE
Google map interface for editing coverage metadata for file types and timeseries resource

### ISSUE THREAD

**Issue by pkdash** [hydroshare/hydroshare#1966]
This needs to be implemented as a common html component (preferably as a modal dialog) that can be used in editing coverage metadata on multiple places on a single page.

**Comment by mauriel133** [comment]
Implemented. Putting documentation below for usage and future reference:

## Usage

### Include the HTML in your template:
`{% include "resource-landing-page/coordinates_picker.html" %}`

### Have a container with all the input fields. 

This container should have:
- The class `"hs-coordinates-picker"` if the elements are to be instantiated on page load. Otherwise see below for how to instantiate dynamically.
- The attribute `"data-coordinates-type"` which can be `"point"` or `"rectangle"`
- The input elements contained within (no matter how nested).

The input elements should have:
- The attribute `"data-map-item"` which can be:
-- for a point: `"longitude"` and `"latitude"`
-- for a rectangle: `"northlimit"`,` "eastlimit"`, `"southlimit"` and `"westlimit"`

### Examples
Elements can be instantiated when the page loads. The script will search for the elements with the class "hs-coordinates-picker" and initiate them. Example:
```
<div class="hs-coordinates-picker" data-coordinates-type="point">
   ...
   <input data-map-item="latitude">
   <input data-map-item="longitude">
   ...
</div>
```

The script will run at page load, find the inputs with the specifications above and append the widget to them. It can be used in multiple places in the same page.

Elements can also be instantiated dynamically. Example:
```
<div id="myCoordinates" data-coordinates-type="point">
   ...
   <input data-map-item="latitude">
   <input data-map-item="longitude">
   ...
</div>

<script>
   $(document).ready(function() {
       $("#myCoordinates").coordinatesPicker();
   });
</script>
```

## Screenshots
_The buttons attached to the input elements_
![image](https://cloud.githubusercontent.com/assets/2448568/24431166/9fe4354a-13d7-11e7-883e-a024a73a4358.png)

_The map modal_
![image](https://cloud.githubusercontent.com/assets/2448568/24431199/cf17a478-13d7-11e7-9db3-64e7a3b19cf5.png)

**Comment by dtarb** [comment]
Closing this.  There are still outstanding issues but they are being addressed in #2330  and #2349
